### DISCUSSION TITLE
Adding disturbances to direct multiple shooting example

### DISCUSSION THREAD

**Discussion by agrace22** [casadi/casadi#3308] [Q&A]
I am interested in adding disturbances to a direct multiple shooting example. Is it possible to add a disturbance vector that is not part of the state vector, x, or the control vector, u, in the direct_multiple_shooting.m example file? For reference, my application will consist of a heat exchanger model with a pre-determined temperature for air flowing into the system that changes with each discrete time step but will not be changed by the optimization/controller.

Thank you in advance for your suggestions.

**Comment by jgillis** [comment]
Yes, these examples are only boilerplates. You can add symbols (MX.sym), create expressions for them and add them to the optimization problem. You probably want to use the keyword 'p' to declare those parameters to be parameters for nlp. If your ode is disturbance-dependent, you may want to add those symbols as an extra input in the definition of `f` and `F`.
